tests/cache/models/samples/upsertBulk.js

```diff
@@ -290,6 +286,67 @@ describe('api::redisEnabled::POST::bulkUpsert ' + path, () => {
     });
   });
 
+  it('bulk upsert should return OK even if every sample in the request ' +
+    'has readonly fields in them', (done) => {
+    api.post(path)
+    .set('Authorization', token)
+    .send([
+      {
+        name: `${tu.namePrefix}Subject|${tu.namePrefix}AspectX`,
+        value: '2',
+        status: 'Invalid',
+      }, {
+        name: `${tu.namePrefix}Subject|${tu.namePrefix}AspectX`,
+        value: '4',
+        statusChangedAt: new Date().toString(),
+      },
+    ])
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        return done(err);
+      }
+      expect(res.body.status).to.equal('OK');
+      return done();
+    });
+  });
+
+  it('samples with read only fields in them should not be upserted', (done) => {
+    api.post(path)
+    .set('Authorization', token)
+    .send([
+      {
+        name: `${tu.namePrefix}Subject|${tu.namePrefix}Aspect1`,
+        value: '10',
+        status: 'Info',
+      },
+      {
+        name: `${tu.namePrefix}Subject|${tu.namePrefix}Aspect2`,
+        value: '10',
+      }
+    ])
+    .expect(constants.httpStatus.OK)
+    .then(() => {
+      /*
+       * the bulk api is asynchronous. The delay is used to give sometime for
+       * the upsert operation to complete
+       */
+      setTimeout(() => {
+        api.get('/v1/samples?name=' +
+        `${tu.namePrefix}Subject|${tu.namePrefix}Aspect*`)
+        .end((err, res) => {
+          if (err) {
+            done(err);
+          }
+          expect(res.body).to.have.length(2);
+          expect(res.body[0].value).to.not.equal('10');
```

Comment:
There is no errorCount field on the GET samples endpoint. The value of res.body[0].value is set by the test that was run before this.